T/TorchCAPI/build_tarballs.jl

```diff
@@ -0,0 +1,114 @@
+using BinaryBuilder, Pkg
+using BinaryBuilderBase: get_addable_spec
+
+include(joinpath(@__DIR__, "..", "..", "fancy_toys.jl"))
+include(joinpath(@__DIR__, "..", "..", "platforms", "cuda.jl"))
+
+name = "TorchCAPI"
+version = v"0.2.0"
+
+torch_version = v"1.10.2"
+
+sources = [
+    GitSource("https://github.com/FluxML/Torch.jl.git", "d1711d716c4993ca25e975aad5f7a638cfa7d7c2"),
+]
+
+script = raw"""
+cmake_extra_args=""
+if [[ $bb_full_target == *cuda* ]]; then
+    # CMake toolchain looks for compiler in CUDA_PATH/bin/nvcc
+    export CUDA_PATH="$prefix/cuda"
+    if [[ $bb_full_target == *cuda+11.3* ]]; then
+        export CUDARTLIB=cudart
+        export cmake_extra_args="\
+            -DCMAKE_TRY_COMPILE_TARGET_TYPE=STATIC_LIBRARY \
+            -DCMAKE_CUDA_COMPILER_ID_RUN=1"
+    fi
+    cmake_extra_args+="\
+        -DUSE_CUDA=ON \
+        -DCUDA_TOOLKIT_ROOT_DIR=$CUDA_PATH"
+else
+    cmake_extra_args+="\
+        -DUSE_CUDA=OFF"
+fi
+
+cd Torch.jl
+install_license LICENSE
+
+cd deps/c_wrapper
+configure() {
+    cmake \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DCMAKE_INSTALL_PREFIX=$prefix \
+        -DCMAKE_TOOLCHAIN_FILE=$CMAKE_TARGET_TOOLCHAIN \
+        $cmake_extra_args \
+        -S . \
+        -B build
+}
+configure || configure || configure
```

Comment:
Sigh.

Reply:
Yeah - it’s just the same silly hack for getting a CUDA 10.2 / 11.3 cmake configure through :-)

Will get rid of that eventually.